lib/modules/datasource/azure-bicep-resource/index.ts

```diff
@@ -0,0 +1,84 @@
+import { cache } from '../../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { BicepTypeIndex } from './schema';
+
+const BICEP_TYPES_INDEX_URL =
+  'https://raw.githubusercontent.com/Azure/bicep-types-az/main/generated/index.json';
+
+export class AzureBicepResourceDatasource extends Datasource {
+  static readonly id = 'azure-bicep-resource';
+
+  override readonly defaultConfig = {
+    commitMessageTopic: 'resource {{depName}}',
+    commitMessageExtra: 'to {{{newVersion}}}',
+    branchTopic: '{{{depNameSanitized}}}-{{{newVersion}}}',
```

Comment:
this will cause anew PR for each version by default. that will probably cat to much noise for most people. we should only open one branch and PR and constantly update that. the separate multiple major option should handle it if a user like to have one PR per version.

Reply:
The best way to group Bicep PRs would be by provider namespace imho.

E.g. `Microsoft.Storage/storageAccounts@2022-09-01`, `Microsoft.Storage/storageAccounts/blobServices@2022-09-01` and `Microsoft.Storage/storageAccounts/blobServices/containers@2022-09-01` should be upgraded together in a single PR. But I don't know how I could configure this without defining a group presets for each namespace provider.

I don't think that grouping all upgrades in a single branch/PR is desirable, because typically in infrastructure as code you don't want to change more than one resource type at the same time to limit possible side effects during deployments.

Reply:
ups, the comment is one line above. i think you should remove branch topic and let renovate handle it.